MediaBrowser.Providers/Lyric/LyricManager.cs

```diff
@@ -0,0 +1,97 @@
+#nullable disable
+
+#pragma warning disable CS1591
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Jellyfin.Extensions;
+using MediaBrowser.Common.Extensions;
+using MediaBrowser.Controller.Entities;
+using MediaBrowser.Controller.Entities.Movies;
+using MediaBrowser.Controller.Entities.TV;
+using MediaBrowser.Controller.Library;
+using MediaBrowser.Controller.Lyrics;
+using MediaBrowser.Controller.Persistence;
+using MediaBrowser.Controller.Providers;
+using MediaBrowser.Controller.Subtitles;
+using MediaBrowser.Model.Configuration;
+using MediaBrowser.Model.Entities;
+using MediaBrowser.Model.Globalization;
+using MediaBrowser.Model.IO;
+using MediaBrowser.Model.Providers;
+using Microsoft.Extensions.Logging;
+
+namespace MediaBrowser.Providers.Lyric
+{
+    public class LyricManager : ILyricManager
+    {
+        private readonly ILogger<LyricManager> _logger;
+        private readonly IFileSystem _fileSystem;
+        private readonly ILibraryMonitor _monitor;
+        private readonly IMediaSourceManager _mediaSourceManager;
+        private readonly ILocalizationManager _localization;
+
+        private ILyricProvider[] _lyricProviders;
+
+        public LyricManager(
+            ILogger<LyricManager> logger,
+            IFileSystem fileSystem,
+            ILibraryMonitor monitor,
+            IMediaSourceManager mediaSourceManager,
+            ILocalizationManager localizationManager)
+        {
+            _logger = logger;
+            _fileSystem = fileSystem;
+            _monitor = monitor;
+            _mediaSourceManager = mediaSourceManager;
+            _localization = localizationManager;
+        }
+
+        /// <inheritdoc />
+        public void AddParts(IEnumerable<ILyricProvider> lyricProviders)
```

Comment:
I would honestly prefer not having AddParts

Reply:
What's the alternative? Can you point me to an example that doesn't use AddParts?

Reply:
We don't have any examples I think. But here's one:

https://github.com/jellyfin/jellyfin/blob/104e36f2f9c6440a7547a4c76d80a69d5af84eea/Jellyfin.Server/CoreAppHost.cs#L89-L92 could be rewritten to 

```
            foreach (var type in GetExportTypes<IWebSocketListener>())
            {
                serviceCollection.AddSingleton(typeof(IWebSocketListener), type);
            }

Reply:
😆  Oh, so like the comment above that section says 

>"// TODO search the assemblies instead of adding them manually?"

I'd prefer to not change that logic as post of this PR since it's not directly part of adding Lyric functionality. It sounds like a good followup for a cleanup PR though.

Reply:
Is it not directly part of it? Instead of using an `AddParts` method, you can scan the assemblies for `ILyricProvider` and add them to the servicecollection. Then put `IEnumerable<ILyricProvider>` in the LyricManager's constructor.

Reply:
Oh, I misunderstood. I thought you were asking me to rework all the serviceCollection.AddSingleton calls as you showed. Which would be touching code unrelated to Lyrics.

After my morning coffee, I now understand you were creating an example for me. 🤦